<?xml version="1.0"?>
<launch>

  <node name="rviz_mobile" pkg="rviz" type="rviz" args="-d $(find ekf_landing)/../rviz_mobile.rviz"/>
  <node name="rviz_drone" pkg="rviz" type="rviz" args="-d $(find ekf_landing)/../rviz_drone.rviz"/>
  <node name="visualizer_for_rviz" pkg="ekf_landing" type="visualizer.py" output="screen"/>
  <node name="tf_broadcaster" pkg="ekf_landing" type="tf_broadcaster.py" output="screen"/>


  <param name="/depth_max_range" value="6.0"/>
  <param name="/depth_fx" value="382.31353296979177"/>
  <param name="/depth_fy" value="383.37247938147743"/>
  <param name="/depth_cx" value="318.8638686003325"/>
  <param name="/depth_cy" value="234.48338338382933"/>

  <param name="/depth_topic" value="/uav/camera/aligned_depth_to_color/image_raw"/>
  <param name="/pcl_topic" value="/converted_pcl"/>
  <param name="/pcl_base" value="cam_link"/>
  <param name="/body_base" value="base_link"/>
  <param name="/agg_pcl_base" value="map"/>

  <node name="ekf_landing_node" pkg="ekf_landing" type="ekf_landing_node" output="screen"/>
  <!-- launch-prefix="gdb -ex run --> <!--args" -->


<!-- edit directly in the python file -->
  <!-- <param name="/backend" value="cv2.dnn.DNN_BACKEND_CUDA"/> -->
  <!-- <param name="/target" value="cv2.dnn.DNN_TARGET_CUDA"/> -->

  <param name="/inference_rate" value="10"/>
  <param name="/img_in_topic" value="/uav/camera/color/image_raw/compressed"/>
  <param name="/img_out" value="True"/>
  <param name="/img_out_topic" value="/detected"/>
  <param name="/bbox_out_topic" value="/bboxes"/>
  <param name="/confidence_threshold" value="0.4"/>
  <param name="/nms_threshold" value="0.4"/>
  <param name="/class_file" value="$(find ekf_landing)/scripts/obj.names"/>
  <param name="/weight_file" value="$(find ekf_landing)/scripts/yolov4-tiny-3l-jackal_final.weights"/>
  <param name="/cfg_file" value="$(find ekf_landing)/scripts/yolov4-tiny-3l-jackal.cfg"/>

  <param name="/yolo_kalman" value="true"/>
  <param name="/uwb_kalman" value="true"/>
  <node name="yolo_ros_detector" pkg="ekf_landing" type="ros_opencv_dnn.py" output="screen"/>

</launch>
